lib/flappi/builder_factory.rb

```diff
@@ -76,12 +76,14 @@ def self.build_and_respond(controller)
       end
 
       # Merge in default values where one is defined and we don't have an actual parameter
-      controller.params.merge! Hash[controller.endpoint_info[:params].select do |defined_param|
-        !controller.params.key?(defined_param[:name]) && defined_param[:default]
-      end.
-      map do |defined_param|
-        [defined_param[:name], defined_param[:default]]
-      end]
+      controller.params.merge! Hash[
+        controller.endpoint_info[:params].select do |defined_param|
+          param = controller.params.dig(defined_param[:name])
```

Comment:
Kind of true, guess I had nested params in mind. But we don't support nor have them yet, right?